activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -387,6 +387,23 @@ def order!(*args) # :nodoc:
       self
     end
 
+    # Allows to specify an order by a specific set of values. Depending on your
+    # adapter this will either use a CASE statement or a built-in function.
+    #
+    #   User.in_order_of(:id, [1, 5, 3])
+    #   # SELECT "users".* FROM "users" ORDER BY FIELD("users"."id", 1, 5, 3)
+    #
+    def in_order_of(column, values)
+      klass.disallow_raw_sql!([column], permit: connection.column_name_with_order_matcher)
+
+      references = column_references([column])
+      self.references_values |= references unless references.empty?
```

Comment:
I tend to find double-negatives harder to read. How about this:

```suggestion
      self.references_values |= references if references.present?
```

Also, did you mean to do a bitwise or here?

Reply:
I'd prefer not using `present?` here as that would work for `nil` values which we don't expect that to be here.

`|` on an array isn't bitwise, it's `union`, which is effectively concat and unique.